src/controllers/loginController.js

```diff
@@ -0,0 +1,72 @@
+let passport = require('passport')
+
+let db = require('../db')
+let ApiError = require('../lib/apiErrors')
+let $ = require('../lib/controllerHelpers')
+
+exports.postLogin = (req, res, next) => {
+  // req.assert('email', 'Email is not valid').isEmail()
+  // req.assert('password', 'Password cannot be blank').notEmpty()
+
+  let errors = req.validationErrors()
+  if (errors) {
+    return next(new ApiError.FailedToLogin(errors))
+  }
+
+  passport.authenticate('local', (err, user, info) => {
+    if (!user || err) {
+      return next(new ApiError.FailedToLogin(err))
+    }
+
+    req.logIn(user, (loginErr) => {
+      if (loginErr) {
+        return next(new ApiError.FailedToLogin(loginErr))
+      }
+      res.json({
+        state: req.isAuthenticated(),
+        user: user
+      })
+    })
+  })(req, res, next)
```

Comment:
Yeah this is a weird little situation based on how HTTP requests are handled in Express middleware. It's actually a bit complicated to explain fully, but I'll add a little note that will hopefully provide a bit of light.